.github/workflows/check-lib.yml

```diff
@@ -0,0 +1,272 @@
+# Checks performed on our common library crates.
+name: Library
+
+on: [push, pull_request]
+
+env:
+  # Use `haswell` instead of `native` due to some GitHub Actions runners not
+  # supporting some `avx512` instructions.
+  RUSTFLAGS: -C target-cpu=haswell
+
+jobs:
+  # Run MSRV first to save Actions time if the code doesn't compile at all.
+  MSRV:
+    name: Minimum supported Rust version
+    runs-on: ubuntu-latest
+    env:
+      minrust: 1.53
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain (${{ env.minrust }})
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.minrust }}
+          profile: minimal
+          override: true
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Output processor info
+        run: cat /proc/cpuinfo
+
```

Comment:
Is this still useful?
```suggestion
```

Reply:
I'd like to keep it around in case any changes to the runner machines happen